packages/builders/__tests__/components/textInput.test.ts

```diff
@@ -0,0 +1,124 @@
+} from '../../src/components/textInput/Assertions';
+import { TextInputComponent } from '../../src/components/textInput/TextInput';
+
+const superLongStr = 'a'.repeat(5000);
+
+const textInputComponent = () => new TextInputComponent();
+
+describe('Text Input Components', () => {
+	describe('Assertion Tests', () => {
+		test('GIVEN valid label THEN validator does not throw', () => {
+			expect(() => labelValidator.parse('foobar')).not.toThrowError();
+		});
+
+		test('GIVEN invalid label THEN validator does throw', () => {
+			expect(() => labelValidator.parse(24)).toThrowError();
+			expect(() => labelValidator.parse(undefined)).toThrowError();
+		});
+
+		test('GIVEN valid style THEN validator does not throw', () => {
+			expect(() => textInputStyleValidator.parse(TextInputStyle.Paragraph)).not.toThrowError();
+		});
+
+		test('GIVEN invalid style THEN validator does throw', () => {
+			expect(() => textInputStyleValidator.parse(24)).toThrowError();
+		});
+
+		test('GIVEN valid min length THEN validator does not throw', () => {
+			expect(() => minLengthValidator.parse(10)).not.toThrowError();
+		});
+
+		test('GIVEN invalid min length THEN validator does throw', () => {
+			expect(() => minLengthValidator.parse(-1)).toThrowError();
+		});
+
+		test('GIVEN valid max length THEN validator does not throw', () => {
+			expect(() => maxLengthValidator.parse(10)).not.toThrowError();
+		});
+
+		test('GIVEN invalid min length THEN validator does throw', () => {
+			expect(() => maxLengthValidator.parse(4001)).toThrowError();
+		});
+
+		test('GIVEN valid value THEN validator does not throw', () => {
+			expect(() => valueValidator.parse('foobar')).not.toThrowError();
+		});
+
+		test('GIVEN invalid value THEN validator does not throw', () => {
+			expect(() => valueValidator.parse(superLongStr)).toThrowError();
+		});
+
+		test('GIVEN valid placeholder THEN validator does not throw', () => {
+			expect(() => placeholderValidator.parse('foobar')).not.toThrowError();
+		});
+
+		test('GIVEN invalid value THEN validator does not throw', () => {
+			expect(() => placeholderValidator.parse(superLongStr)).toThrowError();
+		});
+
+		test('GIVEN valid fields THEN builder does not throw', () => {
+			expect(() => {
+				textInputComponent().setCustomId('foobar').setLabel('test').setStyle(TextInputStyle.Paragraph).toJSON();
+			}).not.toThrowError();
+
+			expect(() => {
+				textInputComponent()
+					.setCustomId('foobar')
+					.setLabel('test')
+					.setMaxLength(100)
+					.setMinLength(1)
+					.setPlaceholder('bar')
+					.setRequired(true)
+					.setStyle(TextInputStyle.Paragraph)
+					.toJSON();
+			}).not.toThrowError();
+		});
+	});
+
+	test('GIVEN invalid fields THEN builder throws', () => {
+		expect(() => textInputComponent().toJSON()).toThrowError();
+		expect(() => {
+			textInputComponent()
+				.setCustomId('test')
+				.setMaxLength(100)
+				.setPlaceholder('hello')
+				.setStyle(TextInputStyle.Paragraph);
+		});
```

Comment:
Expecting what here? toThrow()?
```suggestion
		}).toThrowError();
```